### DISCUSSION TITLE
Custom Validator for Dict Subclass

### DISCUSSION THREAD

**Discussion by ghost** [pydantic/pydantic#3651] [Question]
I have a custom dict subclass that I would like to add validators for. I have tried the usual `__get_validators__` method, but that does not appear to be used when pydantic detects the model field is a subclass of dict. Does pydantic support adding custom validators directly to this class (not in the BaseModel this will be referenced in)? 

```python3
class Collection(dict[KT, VT]):
    ...
```

**Comment by NowanIlfideme** [comment]
I guess pydantic just checks `isinstance(x, dict)`. Consider changing your collection to inherit from `collections.abc.MutableMapping` instead - inheriting from built-in containers is tricky, because there's implementation-specific (e.g. CPython) magic usually going on. Docs: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

**Reply by ghost** [reply]
The problem with MutableMapping is I then need to define all the abstract dunder methods when I really just want it to look like a dict. I ended up using a different definition for typing and runtime. 

```python3
if TYPE_CHECKING:
    class Collection(dict[KT, VT]):
        pass
else:
    class Collection(Generic[KT, VT]):
        ...
```
